### PR TITLE
[1050] Add validation to provider postcode

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
Prevent bad postcode data being sent to downstream systems

### Changes proposed in this pull request
- Update postcode missing validation message
- Add minimum length to postcode validation
- Add maximum length to postcode validation

### Guidance to review
- Try to remove the postcode and publish
- Try less than 7 digits and publish
- Try more than 8 digits and publish

e.g. https://localhost:44364/organisation/t92/details

### CODE REVIEW

File: src/ui/ViewModels/OrganisationViewModel.cs [hunk 1]
```diff
@@ -58,8 +58,18 @@ public OrganisationViewModel()
         [Required(ErrorMessage = "Enter your county")]
         public string Addr4 { get; set; }
 
-        [Required(ErrorMessage = "Enter postcode")]
-        public string Postcode { get; set; }
+        [MinLength(7, ErrorMessage = "Postcode is too short. Enter a postcode in the format ‘SW10 1AA’")]
```
Comment (misaka) [inline src/ui/ViewModels/OrganisationViewModel.cs]: Do we know what the minimum is? The design says 7 is the maximum if we don't include the space.

Comment (dankmitchell) [reply src/ui/ViewModels/OrganisationViewModel.cs]: This is wrong actually. The minimum should be 5

Comment (misaka) [reply src/ui/ViewModels/OrganisationViewModel.cs]: 5 probably has to be the minimum, btw. e.g. `E16DS`

**Comment by timabell** [comment]
[Provider API issues flagged by UCAS: postcodes that are too long](https://trello.com/c/NNKtWy11/1050-provider-api-issues-flagged-by-ucas-postcodes-that-are-too-long)

**Pull request merged**